c7n/actions.py

```diff
@@ -151,7 +151,7 @@ def _run_api(self, cmd, *args, **kw):
         except ClientError as e:
             if (e.response['Error']['Code'] == 'DryRunOperation' and
             e.response['ResponseMetadata']['HTTPStatusCode'] == 412 and
-            'would have succeeded' in e.message):
+            'would have succeeded' == e.response['Error']['Message']):
```

Comment:
I switched to equality since that’s literally the entire message. (Or at least it is today. The real risk here is in branching based on the text of an error. That isn’t guaranteed to stay the same.)

In any case, I can change it back if you prefer. And yes, this is ready for review.